app/src/main/java/org/septa/android/app/notifications/subscription/RefreshAdvertisingId.java

```diff
@@ -0,0 +1,86 @@
+package org.septa.android.app.notifications.subscription;
+
+import android.content.Context;
+import android.os.AsyncTask;
+import android.util.Log;
+
+import com.google.android.gms.ads.identifier.AdvertisingIdClient;
+import com.google.android.gms.common.GooglePlayServicesNotAvailableException;
+import com.google.android.gms.common.GooglePlayServicesRepairableException;
+
+import org.septa.android.app.services.apiinterfaces.SeptaServiceFactory;
+import org.septa.android.app.support.CrashlyticsManager;
+
+import java.io.IOException;
+
+public class RefreshAdvertisingId extends AsyncTask<Object, Object, Void> {
+
+    // tag for logging purposes
+    private static final String TAG = RefreshAdvertisingId.class.getSimpleName();
+
+    private Context context;
+    private Runnable onCancel;
+    private Runnable onPostExecute;
+    private String id = null;
+
+    public RefreshAdvertisingId(Context context, Runnable onCancel, Runnable onPostExecute) {
+        this.context = context;
+        this.onCancel = onCancel;
+        this.onPostExecute = onPostExecute;
+    }
+
+    @Override
+    protected Void doInBackground(Object... voids) {
+
+        AdvertisingIdClient.Info adInfo = null;
+        try {
+            adInfo = AdvertisingIdClient.getAdvertisingIdInfo(context);
+
+        } catch (IOException e) {
+            // Unrecoverable error connecting to Google Play services (e.g.,
+            // the old version of the service doesn't support getting AdvertisingId).
+
+//        } catch (GooglePlayServicesAvailabilityException e) {
+            // Encountered a recoverable error connecting to Google Play services.
+
+        } catch (GooglePlayServicesNotAvailableException e) {
+            // Google Play services is not available entirely.
+        } catch (GooglePlayServicesRepairableException e) {
+            e.printStackTrace();
+        }
+
+        if (adInfo != null) {
```

Comment:
Instead of a null check, why not just move the rest of the method to the try block?